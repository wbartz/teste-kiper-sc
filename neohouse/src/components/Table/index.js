import PropTypes from 'prop-types';
import React from 'react';
import ReactTooltip from 'react-tooltip';
import './index.scss';

const Table = ({ header, lines, onEdit, onRemove, onView }) => {
  return (
    <div className="table">
      <table className="responsive-table striped">
        <thead>
          <tr>
            {header.map((title) => (
              <td key={Math.random()}>{title}</td>
            ))}
          </tr>
        </thead>

        <tbody>
          {lines.length > 0 ? (
            lines.map((line) => (
              <tr key={Math.random()}>
                {Object.keys(line).map((index) => {
                  if (index !== 'id')
                    return <td key={line[index]}>{line[index]}</td>;
                  return null;
                })}
                <td className="table-actions right">
                  <ReactTooltip key={line.id} />
                  <i
                    className="material-icons view"
                    data-tip="Ver Moradores"
                    onClick={() => onView(line.id)}
                  >
                    visibility
                  </i>
                  <i
                    className="material-icons edit"
                    data-tip="Editar Apartamento"
                    onClick={() => onEdit(line.id)}
                  >
                    edit
                  </i>
                  <i
                    className="material-icons remove"
                    data-tip="Remover Apartamento"
                    onClick={() => onRemove(line.id)}
                  >
                    delete
                  </i>
                </td>
              </tr>
            ))
          ) : (
            <tr className="no-results">
              <td colspan={header.length} className="center">
                Nenhum registro encontrado
              </td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  );
};

Table.propTypes = {
  header: PropTypes.oneOfType([PropTypes.array]).isRequired,
  lines: PropTypes.oneOfType([PropTypes.array]).isRequired,
  onEdit: PropTypes.func.isRequired,
  onView: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
};

export default Table;
