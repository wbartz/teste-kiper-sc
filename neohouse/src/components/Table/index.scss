@import '../../styles/variables';


table {
  thead {
    border-bottom: 1px solid $color-lighter-gray;

    td {
      font-size: $h5-font-size;
      font-weight: bold;
    }
  }

  tbody {
    tr.no-results {
      font-size: $h5-font-size;
      font-style: italic;
    }

    .table-actions {
      width: 100px;
      display: flex;
      text-align: right;
      justify-content: space-between;

      i {
        cursor: pointer;
      }

      .view {
        color: $color-secondary;
      }

      .edit {
        color: $color-warning;
      }

      .remove {
        color: $color-error;
      }
    }
  }
}