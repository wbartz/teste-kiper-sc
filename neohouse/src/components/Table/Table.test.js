import { shallow } from 'enzyme';
import React from 'react';
import Table from '.';

describe('Table', () => {
  const props = {
    header: ['id', 'name'],
    lines: [
      {
        id: 1,
        name: 'teste',
      },
    ],
    onEdit: jest.fn(),
    onView: jest.fn(),
    onRemove: jest.fn(),
  };

  it('render without crashing', () => {
    expect(shallow(<Table {...props} />)).toMatchSnapshot();
  });
});
