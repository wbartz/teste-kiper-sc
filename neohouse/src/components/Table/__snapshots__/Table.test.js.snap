// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table render without crashing 1`] = `ShallowWrapper {}`;
